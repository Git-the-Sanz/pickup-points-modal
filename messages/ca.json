{
  "pickupPointsModal.loading": "Carregant…",
  "pickupPointsModal.deliver": "Rebre",
  "pickupPointsModal.pickup": "Recollir",
  "pickupPointsModal.items": "{itemsAmount, plural, =0 {} one {# producte} other {# productes}}",
  "pickupPointsModal.cancelBackList": "Veure tots els punts de recollida",
  "pickupPointsModal.slaOption": "{name} - {price} - {shippingEstimate}",
  "pickupPointsModal.free": "Gratis",
  "pickupPointsModal.selectPickupPoint": "Esculli un punt de recollida",
  "pickupPointsModal.pointDetails": "Detalls del punt de recollida",
  "pickupPointsModal.closeButton": "Tancar",
  "pickupPointsModal.list": "Llista",
  "pickupPointsModal.map": "Mapa",
  "pickupPointsModal.pickupPoint.modify": "Canviar",
  "pickupPointsModal.details": "Veure detalls o canviar el punt de recollida.",
  "pickupPointsModal.searchLocationMap": "Cercar per Lloc, Codi Postal, Barri o Ciutat",
  "pickupPointsModal.productsInPoint": "Productes disponibles en aquest punt",
  "pickupPointsModal.aditionalInfo": "Altres informacions",
  "pickupPointsModal.unavailable": "No disponible",
  "pickupPointsModal.unavailableItemsAmount": "{itemsAmount, plural, =0 {} one {# no disponible} other {# no estan disponibles}}",
  "pickupPointsModal.allItemsAvailable": "Tots els productes disponibles",
  "pickupPointsModal.noneItemsAvailable": "Cap producte disponible",
  "pickupPointsModal.businessHours": "Horaris d'atenció",
  "pickupPointsModal.confirmPoint": "Recollida en aquest punt",
  "pickupPointsModal.price": "{value, plural, =0 {Gratis} other {{formattedPrice}}}",
  "pickupPointsModal.askGeolocation": "Trobi una adreça amb la seva ubicació",
  "pickupPointsModal.marker": "Punt de geolocalització",
  "pickupPointsModal.geolocationDiscover": "Anem a suggerir un punt de recollida a prop teu",
  "pickupPointsModal.geolocationAsk": "Per això, necessitem saber la seva ubicació",
  "pickupPointsModal.geolocationManual": "Cercar per una adreça",
  "pickupPointsModal.geolocationAllow": "Seleccioneu \"Permet \" en l'alerta que apareix al seu navegador",
  "pickupPointsModal.geolocationWaiting": "Esperant la seva autorització",
  "pickupPointsModal.geolocationSearching": "Buscant punts propers",
  "pickupPointsModal.distance": "{distanceValue} km",
  "pickupPointsModal.askGeolocationAccept": "Utilitza la meva ubicació actual",
  "pickupPointsModal.askGeolocationDenied": "No tenim permís per obtenir la vostra ubicació.",
  "pickupPointsModal.errorNotAllowed": "No tenim permís per veure la seva ubicació",
  "pickupPointsModal.errorNotAllowedSubtitle": "Vostè encara pot buscar una adreça",
  "pickupPointsModal.errorNotFound": "No trobem punts de recollida a prop d'aquesta adreça",
  "pickupPointsModal.errorNotFoundSubtitle": "Com més específica sigui la seva recerca, millors seran els resultats!",
  "pickupPointsModal.useSearchBar": "Utilitzeu la cerca per trobar altres punts",
  "pickupPointsModal.errorCouldNotGetLocation": "No trobem el punt exacte d'aquesta adreça",
  "pickupPointsModal.errorCouldNotGetLocationSubtitle": "Busqui en aquest format: Avinguda Diagonal 405",
  "pickupPointsModal.geolocationEmpty": "Trobi punts de recollida propers",
  "pickupPointsModal.geolocationEmptyInstructions": "Com més específica sigui la seva recerca, millors seran els resultats!",
  "pickupPointsModal.postalCodeEmptyInstructions": "Introduïu una adreça",
  "pickupPointsModal.nearTo": "A prop de {address}",
  "pickupPointsModal.weekDay0": "Diumenge",
  "pickupPointsModal.weekDay1": "Dilluns",
  "pickupPointsModal.weekDay2": "Dimarts",
  "pickupPointsModal.weekDay3": "Dimecres",
  "pickupPointsModal.weekDay4": "Dijous",
  "pickupPointsModal.weekDay5": "Divendres",
  "pickupPointsModal.weekDay6": "Dissabte",
  "pickupPointsModal.weekDay1to5": "De dilluns a divendres",
  "pickupPointsModal.hourTo": "a",
  "pickupPointsModal.closed": "Tancat",
  "pickupPointsModal.backToSearch": "Torna a cercar",
  "pickupPointsModal.searchArea": "Cerca aquesta zona",
  "pickupPointsModal.showPickupPointsList": "Veure tots els punts de retirada",
  "pickupPointsModal.bestResults": "Els millors resultats",
  "pickupPointsModal.resultsOrderedByDistance": "Resultats ordenats per distància"
}
