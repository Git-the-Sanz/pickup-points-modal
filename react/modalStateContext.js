import React from 'react'

export const ModalStateContext = React.createContext()

export function injectState(Component) {
  return function StateInjectedComponent(props) {
    return (
      <ModalStateContext.Consumer>
        {({
          activeState,
          activeSidebarState,
          externalPickupPoints,
          geolocationStatus,
          isSearching,
          lastState,
          lastSidebarState,
          lastMapCenterLatLng,
          logisticsInfo,
          pickupOptions,
          pickupPoints,
          selectedPickupPoint,
          setGeolocationStatus,
          setActiveState,
          setActiveSidebarState,
          setAskForGeolocation,
          setSelectedPickupPoint,
          setMapCenterLatLng,
          shouldSearchArea,
          setShouldSearchArea,
        }) => (
          <Component
            {...props}
            activeState={activeState}
            activeSidebarState={activeSidebarState}
            externalPickupPoints={externalPickupPoints}
            geolocationStatus={geolocationStatus}
            isSearching={isSearching}
            lastState={lastState}
            lastSidebarState={lastSidebarState}
            lastMapCenterLatLng={lastMapCenterLatLng}
            logisticsInfo={logisticsInfo}
            selectedPickupPoint={selectedPickupPoint}
            pickupOptions={pickupOptions}
            pickupPoints={pickupPoints}
            setGeolocationStatus={setGeolocationStatus}
            setActiveState={setActiveState}
            setActiveSidebarState={setActiveSidebarState}
            setAskForGeolocation={setAskForGeolocation}
            setMapCenterLatLng={setMapCenterLatLng}
            setSelectedPickupPoint={setSelectedPickupPoint}
            shouldSearchArea={shouldSearchArea}
            setShouldSearchArea={setShouldSearchArea}
          />
        )}
      </ModalStateContext.Consumer>
    )
  }
}
