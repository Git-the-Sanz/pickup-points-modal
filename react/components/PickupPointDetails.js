import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { injectIntl, intlShape } from 'react-intl'
import { translate } from '../utils/i18nUtils'

import {
  getUnavailableItemsByPickup,
  getItemsWithPickupPoint,
  formatBusinessHoursList,
} from '../utils/pickupUtils'

import PickupPointInfo from './PickupPointInfo'
import ProductItems from './ProductItems'
import PickupPointHour from './PickupPointHour'
import Button from './Button'

import styles from './PickupPointDetails.css'
import { LIST } from '../constants'
import { injectState } from '../modalStateContext'
import { updateShippingData } from '../fetchers'

class PickupPointDetails extends Component {
  constructor(props) {
    super(props)

    this.state = {
      unavailableItems: getUnavailableItemsByPickup(
        props.items,
        props.logisticsInfo,
        props.selectedPickupPoint,
        props.sellerId
      ),
      items: getItemsWithPickupPoint(
        props.items,
        props.logisticsInfo,
        props.selectedPickupPoint,
        props.sellerId
      ),
      pickupPointInfo:
        props.selectedPickupPoint.pickupStoreInfo || props.selectedPickupPoint,
    }
  }

  componentDidMount() {
    if (!this.props.selectedPickupPoint) {
      this.props.setActiveSidebarState(LIST)
    }
  }

  handleBackButtonClick = () => {
    this.props.setActiveSidebarState(LIST)
  }

  handleConfirmButtonClick = () => {
    updateShippingData(this.props.logisticsInfo, this.props.selectedPickupPoint)
    this.props.handleClosePickupPointsModal()
  }

  render() {
    const {
      selectedPickupPoint,
      selectedRules,
      isSelectedSla,
      sellerId,
      shouldUseMaps,
      intl,
      storePreferencesData,
      logisticsInfo,
    } = this.props

    const { unavailableItems, items, pickupPointInfo } = this.state

    const businessHours =
      !pickupPointInfo ||
      !pickupPointInfo.businessHours ||
      pickupPointInfo.businessHours.length === 0
        ? null
        : formatBusinessHoursList(pickupPointInfo.businessHours)

    const hasAditionalInfo =
      selectedPickupPoint.pickupStoreInfo &&
      selectedPickupPoint.pickupStoreInfo.additionalInfo

    const shouldShowSelectPickupButton =
      selectedPickupPoint && selectedPickupPoint.pickupStoreInfo

    return (
      <div className={`${styles.modalDetails} pkpmodal-details`}>
        <div className={`${styles.modalDetailsTop} pkpmodal-details-top`}>
          <button
            className={`${
              styles.modalDetailsBackLnk
            } pkpmodal-details-back-lnk btn btn-link`}
            onClick={this.handleBackButtonClick}
            type="button">
            <i
              className={`${
                styles.iconBackPickupPointsList
              } pkpmodal-icon-back-pickup-points-list icon-angle-left`}
            />
            {translate(intl, 'cancelBackList')}
          </button>
        </div>

        <div className={`${styles.modalDetailsMiddle} pkpmodal-details-middle`}>
          <div className={`${styles.modalDetailsStore} pkpmodal-details-store`}>
            <PickupPointInfo
              shouldUseMaps={shouldUseMaps}
              isSelected={isSelectedSla}
              items={this.props.items}
              logisticsInfo={logisticsInfo}
              pickupPoint={selectedPickupPoint}
              selectedRules={selectedRules}
              sellerId={sellerId}
              storePreferencesData={storePreferencesData}
            />
          </div>

          <div className={`${styles.modalDetailsInfo} pkpmodal-details-info`}>
            <div
              className={`${styles.modalDetailsGroup} pkpmodal-details-group`}>
              <h3
                className={`${
                  styles.modalDetailsInfoTitle
                } title pkpmodal-details-info-title`}>
                {translate(intl, 'productsInPoint')}
              </h3>
              {items && <ProductItems items={items} />}
              {unavailableItems && (
                <ProductItems isAvailable={false} items={unavailableItems} />
              )}
            </div>
            {hasAditionalInfo && (
              <div
                className={`${
                  styles.modalDetailsGroup
                } pkpmodal-details-group`}>
                <h3
                  className={`${
                    styles.modalDetailsInfoTitle
                  } pkpmodal-details-info-title`}>
                  {translate(intl, 'aditionalInfo')}
                </h3>
                {selectedPickupPoint.pickupStoreInfo.additionalInfo}
              </div>
            )}

            {businessHours && (
              <div
                className={`${
                  styles.modalDetailsGroup
                } pkpmodal-details-group`}>
                <h3
                  className={`${
                    styles.modalDetailsInfoTitle
                  } pkpmodal-details-info-title`}>
                  {translate(intl, 'businessHours')}
                </h3>
                <table
                  className={`${
                    styles.modalDetailsHours
                  } pkpmodal-details-hours`}>
                  <tbody>
                    {businessHours.map((day, i) => {
                      return (
                        <tr key={i}>
                          <td
                            className={`${
                              styles.modalDetailsHoursDay
                            } pkpmodal-details-hours-day`}>
                            {translate(intl, `weekDay${day.number}`)}
                          </td>
                          {day.closed ? (
                            <td
                              className={`${
                                styles.modalDetailsHoursClosed
                              } pkpmodal-details-hours-closed`}>
                              {translate(intl, 'closed')}
                            </td>
                          ) : (
                            <td
                              className={`${
                                styles.modalDetailsHoursRange
                              } pkpmodal-details-hours-range`}>
                              <PickupPointHour time={day.openingTime} />{' '}
                              {translate(intl, 'hourTo')}{' '}
                              <PickupPointHour time={day.closingTime} />
                            </td>
                          )}
                        </tr>
                      )
                    })}
                  </tbody>
                </table>
              </div>
            )}
          </div>
        </div>

        {shouldShowSelectPickupButton && (
          <div
            className={`${styles.modalDetailsBottom} pkpmodal-details-bottom`}>
            <Button
              id={`confirm-pickup-${selectedPickupPoint.id
                .replace(/[^\w\s]/gi, '')
                .split(' ')
                .join('-')}`}
              kind="primary"
              large
              moreClassName={`${
                styles.modalDetailConfirmBtn
              } pkpmodal-details-confirm-btn`}
              onClick={this.handleConfirmButtonClick}
              title={translate(intl, 'confirmPoint')}
            />
          </div>
        )}
      </div>
    )
  }
}

PickupPointDetails.propTypes = {
  handleChangeActiveSLAOption: PropTypes.func.isRequired,
  handleClosePickupPointsModal: PropTypes.func.isRequired,
  intl: intlShape.isRequired,
  isSelectedSla: PropTypes.bool,
  items: PropTypes.array.isRequired,
  logisticsInfo: PropTypes.array.isRequired,
  onClickPickupModal: PropTypes.func,
  selectedRules: PropTypes.object.isRequired,
  selectedPickupPoint: PropTypes.object.isRequired,
  sellerId: PropTypes.string,
  storePreferencesData: PropTypes.object.isRequired,
  setActiveSidebarState: PropTypes.func.isRequired,
  shouldUseMaps: PropTypes.bool.isRequired,
}

export default injectState(injectIntl(PickupPointDetails))
