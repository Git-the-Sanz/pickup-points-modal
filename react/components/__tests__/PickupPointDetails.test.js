import React from 'react'
import PickupPointDetails from '../PickupPointDetails'

import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import { IntlProvider } from 'react-intl'
import renderer from 'react-test-renderer'
import { addValidation } from '@vtex/address-form'
import BRA from '@vtex/address-form/lib/country/BRA'
import { PICKUP, DELIVERY, PICKUP_IN_STORE, SIDEBAR } from '../../constants'
import messages from '../../../messages/en.json'
import { ModalStateContext } from '../../modalStateContext'
jest.mock('../../utils/Images', () => ({
  fixImageUrl: () => 'teste.png',
}))
describe('PickupPointDetails', () => {
  let state,
    store,
    props,
    handleChangeActiveSLAOption,
    handleClosePickupPointsModal,
    togglePickupDetails,
    setActiveSidebarState,
    modalState

  const address = {
    addressType: 'residential',
    receiverName: null,
    addressId: '10',
    postalCode: '22222222',
    city: 'Rio de janeiro',
    state: 'RJ',
    country: 'BRA',
    street: 'Av das Américas',
    number: '',
    neighborhood: 'Barra da Tijuca',
    complement: 'Loja Barra da Tijuca',
    reference: null,
    geoCoordinates: [],
    addressQuery: 'query',
  }

  beforeEach(() => {
    handleChangeActiveSLAOption = jest.fn()
    handleClosePickupPointsModal = jest.fn()
    togglePickupDetails = jest.fn()
    setActiveSidebarState = jest.fn()

    modalState = {
      activeState: SIDEBAR,
      shouldUseMaps: true,
      setActiveSidebarState,
      shouldUseMaps: false,
      setShouldSearchArea: jest.fn(),
      setSelectedPickupPoint: jest.fn(),
      selectedPickupPoint: {
        name: 'test',
        price: 100,
        shippingEstimate: '1bd',
        pickupStoreInfo: {
          friendlyName: 'test',
          address: {
            geoCoordinates: [123, 123],
          },
        },
        deliveryChannel: PICKUP_IN_STORE,
        id: '1',
      },
      logisticsInfo: [
        {
          itemIndex: 0,
          deliveryChannels: [{ id: DELIVERY }],
          selectedSla: '2',
          slas: [
            {
              name: 'test',
              price: 100,
              shippingEstimate: '1bd',
              deliveryChannel: PICKUP_IN_STORE,
              id: '1',
              pickupStoreInfo: {
                friendlyName: 'test',
                address,
              },
            },
            {
              name: 'test',
              price: 100,
              shippingEstimate: '1bd',
              deliveryChannel: PICKUP_IN_STORE,
              id: '2',
              pickupStoreInfo: {
                friendlyName: 'test',
                address,
              },
            },
          ],
        },
        {
          itemIndex: 1,
          deliveryChannels: [{ id: PICKUP_IN_STORE }],
          selectedSla: '2',
          slas: [
            {
              name: 'test',
              price: 100,
              shippingEstimate: '1bd',
              deliveryChannel: PICKUP_IN_STORE,
              id: '1',
              pickupStoreInfo: {
                friendlyName: 'test',
                address,
              },
            },
            {
              name: 'test',
              price: 100,
              shippingEstimate: '1bd',
              deliveryChannel: PICKUP_IN_STORE,
              id: '2',
              pickupStoreInfo: {
                friendlyName: 'test',
                address,
              },
            },
          ],
        },
        {
          itemIndex: 2,
          deliveryChannels: [{ id: PICKUP_IN_STORE }],
          selectedSla: '2',
          slas: [
            {
              name: 'test',
              price: 100,
              shippingEstimate: '1bd',
              deliveryChannel: PICKUP_IN_STORE,
              id: '1',
              pickupStoreInfo: {
                friendlyName: 'test',
                address,
              },
            },
            {
              name: 'test',
              price: 100,
              shippingEstimate: '1bd',
              deliveryChannel: PICKUP_IN_STORE,
              id: '3',
              pickupStoreInfo: {
                friendlyName: 'test',
                address,
              },
            },
          ],
        },
      ],
    }

    state = {
      pickup: {
        activeSellerId: '1',
        pickupPointId: '2',
        pickupOptions: [
          {
            name: 'test',
            price: 100,
            shippingEstimate: '1bd',
            pickupStoreInfo: {
              friendlyName: 'test',
              address: {
                geoCoordinates: [123, 123],
              },
            },
            deliveryChannel: PICKUP_IN_STORE,
            id: '1',
          },
          {
            name: 'test',
            price: 100,
            shippingEstimate: '1bd',
            pickupStoreInfo: {
              friendlyName: 'test',
              address: {
                geoCoordinates: [123, 123],
              },
            },
            deliveryChannel: PICKUP_IN_STORE,
            id: '2',
          },
        ],
      },
      orderForm: {
        googleMapsKey: '1234',
        clientPreferencesData: {
          locale: 'pt-BR',
        },
        items: [
          {
            name: 'test',
            imageUrl: 'test.png',
            uniqueId: 'xablau',
            seller: '1',
          },
          {
            name: 'test2',
            imageUrl: 'test2.png',
            uniqueId: 'xablau2',
            seller: '1',
          },
          {
            name: 'test3',
            imageUrl: 'test2.png',
            uniqueId: 'xablau3',
            seller: '2',
          },
        ],
        activeTab: PICKUP,
        storePreferencesData: {
          countryCode: 'BRA',
          currencyCode: 'BRL',
          currencySymbol: 'R$',
          currencyFormatInfo: {
            currencyDecimalDigits: 2,
            currencyDecimalSeparator: ',',
            currencyGroupSeparator: '.',
            currencyGroupSize: 3,
            startsWithCurrencySymbol: true,
          },
        },
        shippingData: {
          address: addValidation({
            addressType: 'residential',
            receiverName: null,
            addressId: '10',
            postalCode: '22222222',
            city: 'Rio de janeiro',
            state: 'RJ',
            country: 'BRA',
            street: 'Av das Américas',
            number: '',
            neighborhood: 'Barra da Tijuca',
            complement: 'Loja Barra da Tijuca',
            reference: null,
            geoCoordinates: [],
            addressQuery: 'query',
          }),
          logisticsInfo: [
            {
              itemIndex: 0,
              deliveryChannels: [{ id: DELIVERY }],
              selectedSla: '2',
              slas: [
                {
                  name: 'test',
                  price: 100,
                  shippingEstimate: '1bd',
                  deliveryChannel: PICKUP_IN_STORE,
                  id: '1',
                  pickupStoreInfo: {
                    friendlyName: 'test',
                    address,
                  },
                },
                {
                  name: 'test',
                  price: 100,
                  shippingEstimate: '1bd',
                  deliveryChannel: PICKUP_IN_STORE,
                  id: '2',
                  pickupStoreInfo: {
                    friendlyName: 'test',
                    address,
                  },
                },
              ],
            },
            {
              itemIndex: 1,
              deliveryChannels: [{ id: PICKUP_IN_STORE }],
              selectedSla: '2',
              slas: [
                {
                  name: 'test',
                  price: 100,
                  shippingEstimate: '1bd',
                  deliveryChannel: PICKUP_IN_STORE,
                  id: '1',
                  pickupStoreInfo: {
                    friendlyName: 'test',
                    address,
                  },
                },
                {
                  name: 'test',
                  price: 100,
                  shippingEstimate: '1bd',
                  deliveryChannel: PICKUP_IN_STORE,
                  id: '2',
                  pickupStoreInfo: {
                    friendlyName: 'test',
                    address,
                  },
                },
              ],
            },
            {
              itemIndex: 2,
              deliveryChannels: [{ id: PICKUP_IN_STORE }],
              selectedSla: '2',
              slas: [
                {
                  name: 'test',
                  price: 100,
                  shippingEstimate: '1bd',
                  deliveryChannel: PICKUP_IN_STORE,
                  id: '1',
                  pickupStoreInfo: {
                    friendlyName: 'test',
                    address,
                  },
                },
                {
                  name: 'test',
                  price: 100,
                  shippingEstimate: '1bd',
                  deliveryChannel: PICKUP_IN_STORE,
                  id: '3',
                  pickupStoreInfo: {
                    friendlyName: 'test',
                    address,
                  },
                },
              ],
            },
          ],
        },
      },
    }
    store = {
      subscribe: jest.fn(),
      dispatch: jest.fn(),
      getState: jest.fn(() => state),
    }

    props = {
      sellerId: '1',
      items: [
        {
          seller: '1',
          name: 'test',
          imageUrl: 'test.png',
          uniqueId: 'xablau',
        },
        {
          seller: '1',
          name: 'test2',
          imageUrl: 'test2.png',
          uniqueId: 'xablau2',
        },
      ],
      unavailableItems: [
        {
          name: 'test3',
          imageUrl: 'test2.png',
          uniqueId: 'xablau3',
        },
      ],
      isSelectedSla: false,
      unavailableItemsAmount: 1,
      storePreferencesData: {
        countryCode: 'BRA',
        currencyCode: 'BRL',
        currencySymbol: 'R$',
        currencyFormatInfo: {
          currencyDecimalDigits: 2,
          currencyDecimalSeparator: ',',
          currencyGroupSeparator: '.',
          currencyGroupSize: 3,
          startsWithCurrencySymbol: true,
        },
      },
      pickupPointId: '2',
      pickupOptionGeolocations: [[123, 123], [123, 123]],
      googleMapsKey: '1234',
      address: addValidation({
        addressType: 'residential',
        receiverName: null,
        addressId: '10',
        postalCode: '22222222',
        city: 'Rio de janeiro',
        state: 'RJ',
        country: 'BRA',
        street: 'Av das Américas',
        number: '',
        neighborhood: 'Barra da Tijuca',
        complement: 'Loja Barra da Tijuca',
        reference: null,
        geoCoordinates: [],
        addressQuery: 'query',
      }),
      pickupPoint: {
        name: 'test',
        price: 100,
        shippingEstimate: '1bd',
        deliveryChannel: PICKUP_IN_STORE,
        id: '1',
        pickupDistance: 10000,
        pickupStoreInfo: {
          friendlyName: 'test',
          address: {
            addressId: '10',
            addressType: 'residential',
          },
          additionalInfo: 'test aditionalInfo',
        },
      },
      selectedRules: BRA,
      handleChangeActiveSLAOption,
      handleClosePickupPointsModal,
      togglePickupDetails,
      logisticsInfo: [],
      pickupPointInfo: {},
    }
  })

  it('should render self and components', () => {
    const wrapper = renderer
      .create(
        <Provider store={store}>
          <ModalStateContext.Provider value={modalState}>
            <IntlProvider
              locale="pt"
              messages={{ ...messages, ...{ 'country.BRA': 'BRA' } }}>
              <PickupPointDetails {...props} />
            </IntlProvider>
          </ModalStateContext.Provider>
        </Provider>
      )
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('should simulate go back to list of pickups', () => {
    const wrapper = mount(
      <Provider store={store}>
        <ModalStateContext.Provider value={modalState}>
          <IntlProvider
            locale="pt"
            messages={{ ...messages, ...{ 'country.BRA': 'BRA' } }}>
            <PickupPointDetails {...props} />
          </IntlProvider>
        </ModalStateContext.Provider>
      </Provider>
    )

    const backLink = wrapper.find('button.pkpmodal-details-back-lnk')

    backLink.simulate('click')

    expect(setActiveSidebarState.mock.calls).toHaveLength(1)
  })

  it('should simulate confirm a pickupPoint', () => {
    const wrapper = mount(
      <ModalStateContext.Provider value={modalState}>
        <IntlProvider
          locale="pt"
          messages={{ ...messages, ...{ 'country.BRA': 'BRA' } }}>
          <PickupPointDetails {...props} />
        </IntlProvider>
      </ModalStateContext.Provider>
    )

    const confirmButton = wrapper.find('.pkpmodal-details-confirm-btn')

    confirmButton.simulate('click')

    expect(handleClosePickupPointsModal.mock.calls).toHaveLength(1)
  })
})
