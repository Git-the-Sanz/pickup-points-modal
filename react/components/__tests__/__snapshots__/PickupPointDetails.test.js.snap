// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickupPointDetails should render self and components 1`] = `
<div
  className="modalDetails pkpmodal-details"
>
  <div
    className="modalDetailsTop pkpmodal-details-top"
  >
    <button
      className="modalDetailsBackLnk pkpmodal-details-back-lnk btn btn-link"
      onClick={[Function]}
      type="button"
    >
      <i
        className="iconBackPickupPointsList pkpmodal-icon-back-pickup-points-list icon-angle-left"
      />
      See all pickup points
    </button>
  </div>
  <div
    className="modalDetailsMiddle pkpmodal-details-middle"
  >
    <div
      className="modalDetailsStore pkpmodal-details-store"
    >
      <div
        className="pickupPoint pkpmodal-pickup-point"
        id="1"
        onClick={[Function]}
      >
        <div
          className="pickupPointMain pkpmodal-pickup-point-main"
        >
          <div
            className="pickupPointMarkerPostalCode pkpmodal-pickup-point-marker"
          >
            <svg
              className="pkpmodal-pickup-point-marker-image"
              height="32"
              version="1.1"
              viewBox="0 0 48 48"
              width="32"
              x="0px"
              xmlns="http://www.w3.org/2000/svg"
              y="0px"
            >
              <path
                d="M24,1.11224c-9.38879,0-17,7.61115-17,17 c0,10.1424,12.87262,23.22955,16.2149,26.4566c0.44031,0.42517,1.12988,0.42517,1.57025,0C28.12744,41.3418,41,28.25464,41,18.11224 C41,8.72339,33.38879,1.11224,24,1.11224z"
                fill="#0176ff"
                stroke="#ffffff"
              />
              <circle
                cx="24"
                cy="18"
                fill="#FFFFFF"
                r="6"
              />
            </svg>
            <svg
              className="pkpmodal-pickup-point-marker-image"
              height="32"
              version="1.1"
              viewBox="0 0 48 48"
              width="32"
              x="0px"
              xmlns="http://www.w3.org/2000/svg"
              y="0px"
            >
              <path
                d="M24,1.11224c-9.38879,0-17,7.61115-17,17 c0,10.1424,12.87262,23.22955,16.2149,26.4566c0.44031,0.42517,1.12988,0.42517,1.57025,0C28.12744,41.3418,41,28.25464,41,18.11224 C41,8.72339,33.38879,1.11224,24,1.11224z"
                fill="#0176ff"
                stroke="#ffffff"
              />
              <circle
                cx="24"
                cy="18"
                fill="#FFFFFF"
                r="6"
              />
            </svg>
          </div>
          <div
            className="pickupPointInfo pkpmodal-pickup-point-info"
          >
            <p
              className="pickupPointName pkpmodal-pickup-point-name"
            >
              test
            </p>
            <div
              className="pickupPointAddress pkpmodal-pickup-point-address "
            >
              <div
                className="address-summary address-summary-BRA"
              >
                <br
                  className="line1-delimiter"
                />
                <span
                  className="country"
                >
                  BRA
                </span>
              </div>
            </div>
            <span
              className="pickupPointAllAvailable pkpmodal-pickup-point-available"
            >
              All products are available
            </span>
          </div>
        </div>
        <div
          className="pickupPointSlaAvailabilityPostalCode pkpmodal-pickup-point-sla-availability"
        >
          <span
            className="pickupPointPrice pkpmodal-pickup-point-price"
          >
            R$ 1,00
          </span>
          <span
            className="pickupPointSla pkpmodal-pickup-point-sla"
          >
            TranslateEstimate
          </span>
        </div>
      </div>
    </div>
    <div
      className="modalDetailsInfo pkpmodal-details-info"
    >
      <div
        className="modalDetailsGroup pkpmodal-details-group"
      >
        <h3
          className="modalDetailsInfoTitle title pkpmodal-details-info-title"
        >
          Product availability in this pickup point
        </h3>
        <div
          className="productItems pkpmodal-product-items"
        >
          <span
            className="productItem pkpmodal-product-item "
            data-tip="test"
          >
            <img
              alt="test"
              data-tip="test"
              src="teste.png"
            />
            <div
              className="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
              id={undefined}
            />
          </span>
          <span
            className="productItem pkpmodal-product-item "
            data-tip="test2"
          >
            <img
              alt="test2"
              data-tip="test2"
              src="teste.png"
            />
            <div
              className="__react_component_tooltip place-top type-dark "
              data-id="tooltip"
              id={undefined}
            />
          </span>
        </div>
        <div
          className="productItems pkpmodal-product-items"
        />
      </div>
    </div>
  </div>
  <div
    className="modalDetailsBottom pkpmodal-details-bottom"
  >
    <button
      className="modalDetailConfirmBtn pkpmodal-details-confirm-btn btn pkp-modal-btn primary pkp-modal-btn-primary block pkp-modal-btn-block lg pkp-modal-btn-lg "
      disabled={undefined}
      id="confirm-pickup-1"
      onClick={[Function]}
      type="button"
    >
      Pickup in this point
    </button>
  </div>
</div>
`;
